AdfDataflowFilePrettifier/AdfDataflowFilePrettifier.Tests/TestFiles/SampleDataFlow3_WithEscapedCRInDefinition_Pretty.txt

```diff
@@ -78,65 +78,67 @@
-MarkAsUpdate select(mapColumn(
-		each(match(!in(['id_hash','columns_hash','MaxSurrogateKey'],name)))
-	),
-	skipDuplicateMapInputs: true,
-	skipDuplicateMapOutputs: true) ~> DropUnwantedColumns
-DropUnwantedColsInput alterRow(insertIf(true())) ~> MarkAsInsert
-ExistingDimensionTable filter(toInteger(byName('Active')) == 1 &&
-true()) ~> FilterForActive
-UnionAllData sink(allowSchemaDrift: true,
-	validateSchema: false,
-	deletable:false,
-	insertable:true,
-	updateable:true,
-	upsertable:false,
-	keys:[($PrimaryKey)],
-	format: 'table',
-	skipDuplicateMapInputs: true,
-	skipDuplicateMapOutputs: true) ~> DimensionTableSink
+			"script": "
+/// PRETTIFIED SCRIPT START MARKER /// DO NOT ALTER THIS LINE ///
+parameters{
+	PrimaryKey as string ('ID'),
+	Columns as string ('Player,Team,Salary')
+}
+source(allowSchemaDrift: true,
+	validateSchema: false) ~> GenericInput
+source(allowSchemaDrift: true,
+	validateSchema: false,
+	isolationLevel: 'READ_UNCOMMITTED',
+	format: 'table') ~> ExistingDimensionTable
+AddHashInput, AddHashExisting exists(AddHashInput@id_hash == AddHashExisting@id_hash
+	&& AddHashInput@columns_hash == AddHashExisting@columns_hash,
+	negate:true,
+	broadcast: 'auto')~> NewAndUpdatedRows
+GenericInput derive(id_hash = md5(byName($PrimaryKey)),
+		columns_hash = md5(byNames(split($Columns,',')))) ~> AddHashInput
+FilterForActive derive(id_hash = md5(byNames(split($PrimaryKey,','))),
+		columns_hash = md5(byNames(split($Columns,',')))) ~> AddHashExisting
+AddHashExisting aggregate(MaxSurrogateKey = max(toInteger(byName('Key')))) ~> GetMaxSurrogateKey
+NewAndUpdatedRows keyGenerate(output(Key as long),
+	startAt: 1L) ~> AddKey
+AddKey, GetMaxSurrogateKey join(Key == MaxSurrogateKey || true(),
+	joinType:'cross',
+	broadcast: 'right')~> JoinWithMaxSurrogateKey
+JoinWithMaxSurrogateKey derive(Key = Key + MaxSurrogateKey,
+		Active = 1,
+		ActiveStartTime = currentUTC(),
+		ActiveEndTime = toTimestamp(toString(null()))) ~> AddDimensionColumns
+AddHashExisting, NewAndUpdatedRows exists(AddHashExisting@id_hash == AddHashInput@id_hash,
+	negate:false,
+	broadcast: 'auto')~> FilterForUpdatedValues
+FilterForUpdatedValues derive(Active = 0,
+		ActiveEndTime = currentUTC()) ~> UpdateObsolete
+AddDimensionColumns select(mapColumn(
+		each(match(!in(['id_hash','columns_hash','MaxSurrogateKey'],name)))
+	),
+	skipDuplicateMapInputs: true,
+	skipDuplicateMapOutputs: true) ~> DropUnwantedColsInput
+MarkAsInsert, DropUnwantedColumns union(byName: true)~> UnionAllData
+UpdateObsolete alterRow(updateIf(true())) ~> MarkAsUpdate
+MarkAsUpdate select(mapColumn(
+		each(match(!in(['id_hash','columns_hash','MaxSurrogateKey'],name)))
+	),
+	skipDuplicateMapInputs: true,
+	skipDuplicateMapOutputs: true) ~> DropUnwantedColumns
+DropUnwantedColsInput alterRow(insertIf(true())) ~> MarkAsInsert
+ExistingDimensionTable filter(toInteger(byName('Active')) == 1 &&
+true()) ~> FilterForActive
```

Comment:
FYI, this is the location of an example CRLF in the script definition.
The ugly file contains: `"== 1 &&\r\ntrue())"`

Reply:
My *guess* would be that the rule is "if there's a new line in the text that defines a operation expression that was entered by the user, then that new line is (or *can* be) a CRLF"? 

Recall that, overall we have:
* A JSON file, containing a prop that is a string
* That represents a script file, containing a sequence of blocks
* Which represent configured operations of a dataset
* Where some of that configuration consists of multi-line free-text entered by the user.

=:0

Reply:
eesh ... it's even worse :D ... there's an extra bullet point:

* where that free text could legitimately be defining multi-line string values!